### PR TITLE
fix: no more circular dependency, become a good block of libp2p

### LINKED ISSUE #9: Circular dep with libp2p-ipfs-nodejs
This module is a piece of js-libp2p bundles, but at the same time it expects to receive a bundle to operate, forcing us to have to create an instance inside the base class. This is annoying to work with because it forces updating and development to happen through `npm link` with a bundle for changes that require modifying how the bundle gets created. This module should use the internal components and not the bundle itself.

### PR DESCRIPTION
**Pull request by diasdavid**
Fix #9. Just one last thing, fix:

```
 1) KadDHT put - get:
   Uncaught TypeError: pi.id.toB58String is not a function
    at Swarm.dial (node_modules/libp2p-swarm/src/dial.js:22:25)
    at Network.sendRequest (src/network.js:143:20)
    at KadDHT._findPeerSingle (src/private.js:244:17)
```

by leveling up swarm.dial to also support PeerId instead of just PeerInfo.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -32,66 +33,71 @@ class KadDHT {
...
      * k-bucket size, defaults to 20.
      *
      * @type {number}
      */
-    this.kBucketSize = kBucketSize || 20
+    this.kBucketSize = options.kBucketSize || 20
 
     /**
-     * Number of closest peers to return on kBucket search
+     * Number of closest peers to return on kBucket search, default 20
```
Comment (dignifiedquire) [inline src/index.js]: this is not 20 in the code but 6

File: test/kad-dht.spec.js [hunk 2]
```diff
@@ -11,23 +11,26 @@ const timeout = require('async/timeout')
 const retry = require('async/retry')
```
Comment (dignifiedquire) [inline test/kad-dht.spec.js]: 👎 for the rename

File: src/index.js [hunk 3]
```diff
@@ -28,70 +29,74 @@ class KadDHT {
...
    *
-   * @param {Libp2p} libp2p
-   * @param {number} [kBucketSize=20]
-   * @param {Datastore} [datastore=MemoryDatastore]
+   * @param {swarm} Swarm
+   * @param {options} {kBucketSize=20, datastore=MemoryDatastore}
    */
-  constructor (libp2p, kBucketSize, datastore) {
+  constructor (swarm, options) {
+    assert(swarm, 'libp2p-kad-dht requires a instance of kad-dht')
```
Comment (pgte) [inline src/index.js]: don't you mean "libp2p-kad-dht requires a instance of **swarm**"?

**Comment by diasdavid** [comment]
CI is failing due to previously reported bug: https://github.com/libp2p/js-libp2p-kad-dht/issues/8

**Pull request merged**